prov/efa/src/efa_av.c

```diff
@@ -358,97 +357,98 @@ static int efa_conn_implicit_to_explicit(struct efa_av *av,
 	assert(ofi_genlock_held(&av->util_av.lock));
 	assert(ofi_genlock_held(&av->util_av_implicit.lock));
 
-	util_av_entry =
+	/* Get implicit util AV entry and conn */
+	implicit_util_av_entry =
 	ofi_bufpool_get_ibuf(av->util_av_implicit.av_entry_pool, implicit_fi_addr);
 
-	implicit_av_entry = (struct efa_av_entry *) util_av_entry->data;
+	implicit_av_entry = (struct efa_av_entry *) implicit_util_av_entry->data;
 
 	assert(implicit_av_entry);
 	assert(efa_is_same_addr(
 		raw_addr, (struct efa_ep_addr *) implicit_av_entry->ep_addr));
 
-	conn = &implicit_av_entry->conn;
-	assert(conn->fi_addr == FI_ADDR_NOTAVAIL &&
-	       conn->implicit_fi_addr == implicit_fi_addr);
+	implicit_conn = &implicit_av_entry->conn;
+	assert(implicit_conn->fi_addr == FI_ADDR_NOTAVAIL &&
+	       implicit_conn->implicit_fi_addr == implicit_fi_addr);
 
-	ah = conn->ah;
-	assert(ah);
-	shm_fi_addr = implicit_av_entry->conn.shm_fi_addr;
-
-	efa_av_reverse_av_remove(&av->cur_reverse_av_implicit,
-				 &av->prv_reverse_av_implicit, conn);
+	ah = implicit_conn->ah;
 
-	err = ofi_av_remove_addr(&av->util_av_implicit, implicit_fi_addr);
+	/* Create explicit util AV entry and conn */
+	err = ofi_av_insert_addr(&av->util_av, raw_addr, fi_addr);
```

Comment:
It probably doesn't matter, but any reason we choose to insert explicit av addr first and then remove the implicit av addr?